presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/TestMergeAdjacentWindows.java

```diff
@@ -87,37 +87,84 @@ public void testDoesNotFireForDistinctAdjacentWindowSpecifications()
         tester.assertThat(new MergeAdjacentWindows())
                 .on(p ->
                         p.window(
-                                new WindowNode.Specification(ImmutableList.of(p.symbol("a", BIGINT)), ImmutableList.of(), ImmutableMap.of()),
-                                ImmutableMap.of(p.symbol("avg_1", BIGINT), new WindowNode.Function(new FunctionCall(QualifiedName.of("avg"), ImmutableList.of()), signature, frame)),
+                                newWindowNodeSpecification(p, "a"),
+                                ImmutableMap.of(p.symbol("avg_1", BIGINT), newWindowNodeFunction("avg", "a")),
                                 p.window(
-                                        new WindowNode.Specification(ImmutableList.of(p.symbol("b", BIGINT)), ImmutableList.of(), ImmutableMap.of()),
-                                        ImmutableMap.of(p.symbol("sum_1", BIGINT),
-                                                new WindowNode.Function(new FunctionCall(QualifiedName.of("sum"), ImmutableList.of()), signature, frame)),
+                                        newWindowNodeSpecification(p, "b"),
+                                        ImmutableMap.of(p.symbol("sum_1", BIGINT), newWindowNodeFunction("sum", "b")),
                                         p.values(p.symbol("a", BIGINT), p.symbol("b", BIGINT))
                                 )
                         ))
                 .doesNotFire();
     }
 
     @Test
+    public void testDoesNotFireForNonWindowIntermediateNode()
+    {
+        tester.assertThat(new MergeAdjacentWindows())
+                .on(p ->
+                        p.window(
+                                newWindowNodeSpecification(p, "a"),
+                                ImmutableMap.of(p.symbol("avg_1", BIGINT), newWindowNodeFunction("avg", "a")),
```

Comment:
Can you use real use case for plan with Projection preceding WN? E.g. have a look what plan with `lag` function looks like.
Simple put `lag` function in WN and Projection providing arguments for `lag` function, as it would be in real plan.
This would ensure that we test against realistic case.